### PR TITLE
Enable notifications plugins in reporting

### PR DESCRIPTION
**Pull request by yenienserrano**
### Description
This pull request enables the notification plugin services.

<details><summary>Screenshot</summary>

Table

image

Details

image

Edit

image

Create

image

</details> 

### Issues Resolved
- https://github.com/wazuh/wazuh-dashboard/issues/288

### Test

Preconditions

Remove the opensearch-reports-scheduler plugin and install the plugin from the wazuh-indexer-reporting repository.

Steps

1. Configuring an SMTP channel in the notification plugin https://opensearch.org/docs/latest/observing-your-data/notifications/index/#email-as-a-channel-type
2. Create a scheduled report definition
3. Try editing the text of the message and containing {{urlDefinition}} to see the url to generate the report.
4. In the table you should see that the notification is activated.
5. In the detail you should see the name of the channel in the title of the mail and the body of the mail.
6. Wait for the mail to be sent

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: public/components/main/main_utils.tsx [hunk 1]
```diff
@@ -264,3 +272,46 @@ export const generateReportById = async (
       }
     });
 };
+export const sendTestNotificationsMessage = async (
+  id: string,
+  httpClientProps: HttpSetup,
+  item: any
+) => {
+  try {
+    console.log({ id, httpClientProps, item });
```
Comment (guidomodarelli) [inline public/components/main/main_utils.tsx:281]: Remove this
```suggestion
```

File: public/components/main/main_utils.tsx [hunk 2]
```diff
@@ -264,3 +272,46 @@ export const generateReportById = async (
       }
     });
 };
+export const sendTestNotificationsMessage = async (
+  id: string,
+  httpClientProps: HttpSetup,
+  item: any
```
Comment (guidomodarelli) [inline public/components/main/main_utils.tsx:278]: Try to typed this prop

File: public/components/main/reports_table.tsx [hunk 3]
```diff
@@ -95,7 +96,7 @@ export function ReportsTable(props) {
     setShowLoading(e);
   };
 
-  const onDemandDownload = async (id: any) => {
+  const onDemandDownload = async (id: any, item: any) => {
```
Comment (guidomodarelli) [inline public/components/main/reports_table.tsx:99]: Try to typed this prop

**Comment by guidomodarelli** [comment]
Test :green_circle: 

Peek 2024-10-03 14-03

Screenshot_3-10-2024_14620_0 0 0 0

image

File: public/components/main/main_utils.tsx (hunk 2 above)
Comment (yenienserrano) [reply public/components/main/main_utils.tsx:278]: This comes from a prop that already comes from the reporting repository and has as type any

File: public/components/main/reports_table.tsx (hunk 3 above)
Comment (yenienserrano) [reply public/components/main/reports_table.tsx:99]: This comes from a prop that already comes from the reporting repository and has as type any

**Review by guidomodarelli — APPROVED** [review]

**Pull request merged**